src/tools/r2rdump/R2RMethod.cs

```diff
@@ -130,21 +107,163 @@ class R2RMethod
         /// <summary>
         /// The id of the entrypoint runtime function
         /// </summary>
-        public uint EntryPointRuntimeFunctionId { get; }
+        public int EntryPointRuntimeFunctionId { get; }
+
+        /// <summary>
+        /// Maps all the generic parameters to the type in the instance
+        /// </summary>
+        Dictionary<string, GenericElementTypes> _genericParamInstance;
 
-        public R2RMethod(byte[] image, MetadataReader mdReader, NativeArray methodEntryPoints, uint offset, uint rid)
+        [Flags]
+        public enum EncodeMethodSigFlags
         {
-            // get the id of the entry point runtime function from the MethodEntryPoints NativeArray
+            NONE = 0x00,
+            ENCODE_METHOD_SIG_UnboxingStub = 0x01,
+            ENCODE_METHOD_SIG_InstantiatingStub = 0x02,
+            ENCODE_METHOD_SIG_MethodInstantiation = 0x04,
+            ENCODE_METHOD_SIG_SlotInsteadOfToken = 0x08,
+            ENCODE_METHOD_SIG_MemberRefToken = 0x10,
+            ENCODE_METHOD_SIG_Constrained = 0x20,
+            ENCODE_METHOD_SIG_OwnerType = 0x40,
+        };
+
+        public enum GenericElementTypes
+        {
+            __Canon = 0x3e,
+            Void = 0x01,
+            Boolean = 0x02,
+            Char = 0x03,
+            Int8 = 0x04,
+            UInt8 = 0x05,
+            Int16 = 0x06,
+            UInt16 = 0x07,
+            Int32 = 0x08,
+            UInt32 = 0x09,
+            Int64 = 0x0a,
+            UInt64 = 0x0b,
+            Float = 0x0c,
+            Double = 0x0d,
+            String = 0x0e,
+            ValueType = 0x11,
+            Object = 0x1c,
+            Array = 0x1d,
+        };
+
+        /// <summary>
+        /// Extracts the method signature from the metadata by rid
+        /// </summary>
+        public R2RMethod(byte[] image, MetadataReader mdReader, uint rid)
+        {
+            _mdReader = mdReader;
+            NativeCode = new List<RuntimeFunction>();
+
+            // get the method signature from the MethodDefhandle
+            MethodDefinitionHandle methodDefHandle = MetadataTokens.MethodDefinitionHandle((int)rid);
+            MethodDefinition methodDef = mdReader.GetMethodDefinition(methodDefHandle);
+            Name = mdReader.GetString(methodDef.Name);
+            BlobReader signatureReader = mdReader.GetBlobReader(methodDef.Signature);
+
+            DeclaringType = new Stack<string>();
+            TypeDefinition declaringTypeDef = mdReader.GetTypeDefinition(methodDef.GetDeclaringType());
```

Comment:
From the TypeDefinition you should be able to access the `Namespace` property which should be a string representation of the namespace. If you use that + the declaring type + the method name, you should have the fully qualified name.

Reply:
If it's a nested namespace, it uses NamespaceHandle to get the parent namespaces